### PR TITLE
NIAD-141: EDIFACT to FHIR (GP and HA data)

### PR DESCRIPTION
**Pull request by grzegorz-robak**

### CODE REVIEW

**Review by mattd-kainos — DISMISSED** [review]

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java [hunk 1]
```diff
@@ -1,14 +1,53 @@
...
+        patient.setManagingOrganization(managingOrganization);
+        patient.addGeneralPractitioner(generalPractitioner);
+
+        return createParameters(patient);
     }
 
+    private Parameters createParameters(Resource... resources) {
+        Parameters parameters = new Parameters();
+        for (Resource resource : resources) {
+            parameters.addParameter(new Parameters.ParametersParameterComponent().setResource(resource));
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java]: Shouldn't parameters be named? I think the example had a name on it

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java [hunk 2]
```diff
@@ -1,14 +1,53 @@
...
 import org.springframework.stereotype.Component;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.Interchange;
 
 @Component
 public class EdifactToFhirService {
 
-    Parameters convertToFhir(Interchange interchange) {
-        return new Parameters();
+    public Parameters convertToFhir(Interchange interchange) {
+        Patient patient = createPatient("example");
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java]: Do we know what should be the ID of patient resource? Left empty or a random GUID?

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java [hunk 3]
```diff
@@ -1,14 +1,53 @@
...
+        for (Resource resource : resources) {
+            parameters.addParameter(new Parameters.ParametersParameterComponent().setResource(resource));
+        }
+        return parameters;
+    }
+
+    private Reference createGeneralPractitionerReference(Interchange interchange) {
+        String gpId = interchange.getGpNameAndAddress().getIdentifier();
+        return new Reference(
+                new Practitioner().setId(gpId)
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java:45]: Shouldn't this just be a string ref like "Practitioner/<id>" ? The same for organization

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java (hunk 3 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java:45]: Output of test patient looked like that:
````
{
  "resourceType": "Parameters",
  "parameter": [
    {
      "resource": {
        "resourceType": "Patient",
        "identifier": [
          {
            "value": "example id"
          }
        ],
        "generalPractitioner": [
          {
            "reference": "Practitioner/2750922,295"
          }
        ],
        "managingOrganization": {
          "reference": "Organization/XX1"
        }
      }
    }
  ]
}
````

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java (hunk 2 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java]: Patient information will be mapped in another ticket - I just used a placeholder for now

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java (hunk 1 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java]: They can be, but naming works only on value params, so it's useless for resources

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java (hunk 2 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java]: I think it's better to set something like "NOT_SET" value here than "example". Example looks like a remaining from not finished work

Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java]: Updated with value that "stands out" and TODO

**Review by bartek-sarul — APPROVED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java (hunk 1 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java]: Added for clarity

**Review by jaroslaw-wojciechowski — APPROVED** [review]

**Pull request merged**